billing/views.py

```diff
@@ -128,15 +129,32 @@ def subscription_schedule_updated(
                 ),
             )
 
-    def subscription_schedule_released(self, schedule: stripe.Subscription) -> None:
+    def subscription_schedule_released(
+        self, schedule: stripe.SubscriptionSchedule
+    ) -> None:
         subscription = stripe.Subscription.retrieve(schedule["released_subscription"])
-        owner = Owner.objects.get(ownerid=subscription.metadata.get("obo_organization"))
-        requesting_user_id = subscription.metadata.get("obo")
-        plan_service = PlanService(current_org=owner)
+        owners: QuerySet[Owner] = Owner.objects.filter(
+            stripe_subscription_id=subscription.id,
+            stripe_customer_id=subscription.customer,
+        )
+        if not owners.exists():
+            log.error(
+                "Subscription schedule released requested with invalid subscription",
+                extra=dict(
+                    stripe_subscription_id=subscription.id,
+                    stripe_customer_id=subscription.customer,
+                    plan_id=subscription.plan.id,
+                ),
+            )
+            return
+
+        plan_services = [PlanService(current_org=owner) for owner in owners]
```

Comment:
Should we keep a similar syntax like we did for subscription_deleted with a top level `for owner in owners` with all the business logic in there? Just to reduce cognitive load in this file

Reply:
yes this make sense here!